bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/BusinessPartnerBuildService.kt

```diff
@@ -506,13 +506,15 @@ class BusinessPartnerBuildService(
         )
     }
 
-    private fun toEntity(dto: ClassificationDto, partner: LegalEntity): LegalEntityClassification {
-        return LegalEntityClassification(
-            value = dto.value,
-            code = dto.code,
-            type = dto.type,
-            legalEntity = partner
-        )
+    private fun toEntity(dto: ClassificationDto, partner: LegalEntity): LegalEntityClassification? {
```

Comment:
The changes to the nullable DTO fields should not affect the Pool. Please split off a Pool specific DTO for the classification and keep the type there non-nullable. There should be a commin Classification base DTO interface which both Pool and the other modules' DTOs inherit from.